src/extension.ts

```diff
@@ -672,6 +684,154 @@ async function createStream(chatMessages: OpenAIChatMessage[], override: OpenAI.
+		return vscode.workspace.applyEdit(edit);
+	}
+
+	const textEditor = vscode.window.activeTextEditor;
+	if (textEditor === undefined) { return; }
+
+	const document = textEditor.document;
+	const userRange = toOverflowAdjustedRange(textEditor, undefined);
+	const workspaceFolder = vscode.workspace.workspaceFolders![0];
+
+	// List all immediate descendants of the `uri` directory
+
+	let diffs = "";
+	const entries = await vscode.workspace.fs.readDirectory(uri);
+	for (const [name, type] of entries) {
+		if (type == vscode.FileType.File && name.endsWith(".md")) {
+			// Get path relative to the workspace folder
+			let path = vscode.Uri.joinPath(uri, name).fsPath.substring(workspaceFolder.uri.fsPath.length + 1);
+			diffs += `- ${path}\n+ ${path}\n`;
+		}
+	}
+
+	insertMessage(diffs)
+}
+
+async function applyFilePathDiff(selectionOverride?: vscode.Selection) {
+	class Diff {
+		from: string;
+		to: string;
+
+		constructor(from: string, to: string) {
+			this.from = from;
+			this.to = to;
+		}
+	}
+
+	async function insertErrorMessage(message: string) {
+		const edit = new vscode.WorkspaceEdit();
+		const anchorPosition = document.positionAt(document.offsetAt(userRange.end));
+		edit.insert(document.uri, anchorPosition, `\nERROR: ${message}`.replaceAll("\n", toEolString(document.eol)));
+		return vscode.workspace.applyEdit(edit);
+	}
+
+	const textEditor = vscode.window.activeTextEditor;
+	if (textEditor === undefined) { return; }
+
+	if (selectionOverride === undefined) {
+		if (textEditor.selection.isEmpty) { return; }
+	} else if (selectionOverride.isEmpty) { return; }
+
+	const document = textEditor.document;
+	const userRange = toOverflowAdjustedRange(textEditor, selectionOverride);
+	const workspaceFolder = vscode.workspace.workspaceFolders![0];
+	const text = document.getText(userRange);
+
+	// Extract diffs
+
+	let path_from = null, path_to = null;
+	let diff_list: Diff[] = [];
+	for (const line of text.split(/\r?\n/)) {
+		if (line.match(/^\+\s*$/)) {
+			// The line only containing `+` is used to delete the file.
+			path_to = "";
+		} else {
+			const match = line.match(/^(\+|\-)\s*([^\s]+)$/);
+			if (match === null) { continue; }
+
+			const [_, op, path] = match;
+			if (op === '-') {
+				path_from = path;
+			} else {
+				path_to = path;
+			}
+		}
+
+		if (path_from !== null && path_to !== null) {
+			diff_list.push(new Diff(path_from, path_to));
+			path_from = path_to = null;
+		}
+	}
+
+	// Detect errors
+
+	// Incomplete diff
+	if (path_from !== null || path_to !== null) {
+		insertErrorMessage(l10n.t("command.editing.applyFilePathDiff.error.incomplete"));
+		return;
+	}
+
+	// File not found
+	for (const { from } of diff_list) {
+		let reason = await vscode.workspace.fs.stat(vscode.Uri.joinPath(workspaceFolder.uri, from))
+			.then(() => null, (reason) => reason);
+		if (reason !== null) {
+			insertErrorMessage(l10n.t("command.editing.applyFilePathDiff.error.fileNotFound", from));
+			return;
```

Comment:
What do you think about listing all missing files and then stopping the process?
Because in the future users will give the LLM the original "File Path Diff" and error messages and let it do an automatic "File Path Diff" edit.